File: app/src/main/java/com/standoff/bang/bang/BangActivity.java
package com.standoff.bang.bang;

import android.app.Activity;
import android.os.Bundle;

import com.github.nkzawa.emitter.Emitter;
import com.github.nkzawa.socketio.client.IO;
import com.github.nkzawa.socketio.client.Socket;

import org.json.JSONObject;

import java.net.URISyntaxException;

public class BangActivity extends Activity {

    private Socket mSocket;
    {
        try {
            mSocket = IO.socket("http://chat.socket.io");
        } catch (URISyntaxException e) {}
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_bang);
        Emitter.Listener onNewMessage = new Emitter.Listener() {
            @Override
            public void call(Object... args) {
                JSONObject data = (JSONObject) args[0];
                System.out.println(data);
            }
        };
        mSocket.on("new message", onNewMessage);
        mSocket.connect();

    }
}

File: app/src/main/java/com/standoff/bang/bang/BangActivity.kt
package com.standoff.bang.bang

import android.app.Activity
import android.os.Bundle
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_bang.*

/**
 * Created by kelvinhanma on 9/28/17.
 */
class BangActivity : Activity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_bang)
        val network = (application as BangApplication).network
        network.getEvents().subscribe({ event ->
            println(event)
        })
        joinButton.setOnClickListener {
            network.connect()
            Toast.makeText(this, "Joining game", Toast.LENGTH_LONG).show()
        }
    }
}
